boot.go

```diff
@@ -82,8 +82,12 @@ func main() {
 				log.Printf("Starting health check server on port %d", cnf.HealthSrvPort)
 				healthSrvCh := make(chan error)
 				go func() {
-					if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
-						healthSrvCh <- err
+					for{
+						if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
+							healthSrvCh <- err
+							break
+						}
+						log.Print("namespace watcher channel closed.Starting the watch again")
```

Comment:
since we are getting the connection again and starting cleaning of the apps i don't think it would disturb build flow as only when the disk space fills does it have a problem.
What if the connections regularly closes which means we would be restarting the pod regularly and  i think the pod should be restarted only when it is absolute necessity that only a restart will help it.